### PR TITLE
#126: More logging

### PR DESCRIPTION
**Pull request by JakenVeina**

### CODE REVIEW

**Comment by jrusbatch** [comment]
Since these are errors should we return early instead of proceeding with the call to `TrackUserAsync`?

File: Modix.Services/Core/UserTrackingBehavior.cs [hunk 1]
```diff
@@ -69,9 +71,21 @@ private Task OnUserJoinedAsync(IGuildUser guildUser)
             => SelfExecuteOnScopedServiceAsync<IUserService>(x => x.TrackUserAsync(guildUser));
 
         private Task OnGuildMemberUpdatedAsync(IGuildUser oldUser, IGuildUser newUser)
-            => SelfExecuteOnScopedServiceAsync<IUserService>(x => x.TrackUserAsync(newUser));
+            => SelfExecuteOnScopedServiceAsync<IUserService>(x =>
+            {
+                if(newUser.Username == null)
+                    Log.Error($"OnGuildMemberUpdatedAsync:\r\n ~ newUser.Id: {newUser.Id}\r\n ~ newUser.Discriminator: {newUser.Discriminator ?? "null"}\r\n ~ newUser.Nickname: {newUser.Nickname ?? "null"}");
```
Comment (jrusbatch) [inline Modix.Services/Core/UserTrackingBehavior.cs]: Serilog formats the messages for you, so these can be simplified to: `Log.Error("OnGuildMemberUpdatedAsync:\r\n ~ newUser.Id: {UserId}\r\n ~ newUser.Discriminator: {UserDiscriminator}\r\n ~ newUser.Nickname: {UserNickname}", newUser.Id, newUser.Discriminator, newUser.Nickname);`

Comment (JakenVeina) [reply Modix.Services/Core/UserTrackingBehavior.cs]: Well, shit. That's nice.

**Comment by JakenVeina** [comment]
There's a bunch of temp logic in `TrackUserAsync` already. Once we figure out exactly what's causing these messages, and how we can avoid them, we can re-evaluate exactly where the handling should occur.

**Pull request merged**